temp_link_generator.py

```
@@ -0,0 +1,116 @@
+        access_key (str): The access key
+        timestamp (int, optional): Unix timestamp (defaults to current time)
+    
+    Returns:
+        str: The temporary link URL
+    """
+    if timestamp is None:
+        timestamp = int(time.time())
+    
+    # Generate the MD5 hash
+    data = f"{timestamp}{bucket}{key}{access_key}"
+    link_key = hashlib.md5(data.encode('utf-8')).hexdigest()
```

Ruff (0.12.2)

Local variable `link_key` is assigned to but never used

Remove assignment to unused variable `link_key`

(F841)